### PR TITLE
Bulk commit: Implemented more blocks

### PR DESCRIPTION
**Pull request by robertdudaa**
Implemented:
- Sandstone
- SandstoneCut
- SandstoneChiseled
- SandstoneSmooth
- Sand
- Gravel
- Flint (Item)

### CODE REVIEW

File: dragonfly/block/gravel.go [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+// Gravel is a block that is affected by gravity.
+type Gravel struct{}
+
+// BreakInfo ...
+func (g Gravel) BreakInfo() BreakInfo {
+	return BreakInfo{
+		Hardness:    0.6,
+		Harvestable: alwaysHarvestable,
+		Effective:   shovelEffective,
+		Drops:       simpleDrops(),
```
Comment (TwistedAsylumMC) [inline dragonfly/block/gravel.go:12]: This will cause gravel to drop nothing

File: dragonfly/block/sand.go [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+}
+
+// BreakInfo ...
+func (s Sand) BreakInfo() BreakInfo {
+	return BreakInfo{
+		Hardness:    0.5,
+		Harvestable: alwaysHarvestable,
+		Effective: func(t tool.Tool) bool {
+			return t.ToolType() == tool.TypeShovel
+		},
```
Comment (TwistedAsylumMC) [inline dragonfly/block/sand.go:21]: ``shovelEffective`` can be used here

**Comment by TwistedAsylumMC** [comment]
Would also be nice if you could implement gravity to these blocks within *this* pull request

File: dragonfly/item/flint.go [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+	"github.com/df-mc/dragonfly/dragonfly/world"
+)
+
+// Flint is a mineral obtained from gravel.
+type Flint struct {
+	None tool.None
+}
+
+// AttackDamage returns the attack damage of the flint.
+func (f Flint) AttackDamage() float64 {
```
Comment (Sandertv) [inline dragonfly/item/flint.go:14]: This function can be removed, the default is 1 attack damage.

File: dragonfly/item/flint.go [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+	return 1
+}
+
+// MaxCount always returns 64.
+func (f Flint) MaxCount() int {
```
Comment (Sandertv) [inline dragonfly/item/flint.go:19]: This function can be removed, the max count is 64 by default.

File: dragonfly/item/flint.go [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+	return 64
+}
+
+// ToolType ...
+func (f Flint) ToolType() tool.Type {
```
Comment (Sandertv) [inline dragonfly/item/flint.go:24]: Flints aren't tools, they don't need to implement any of the Tool methods.

File: dragonfly/block/sandstone_cut.go [hunk 6]
```diff
@@ -0,0 +1,36 @@
+package block
+
+import "github.com/df-mc/dragonfly/dragonfly/item"
+
+// SandstoneCut is a block which can be found in generated structures.
+type SandstoneCut struct {
```
Comment (Sandertv) [inline dragonfly/block/sandstone_cut.go:6]: CutSandstone would make more sense.

File: dragonfly/block/sandstone_chiseled.go [hunk 7]
```diff
@@ -0,0 +1,36 @@
+package block
+
+import "github.com/df-mc/dragonfly/dragonfly/item"
+
+// SandstoneChiseled is a block which naturally generates in Desert Villages and Desert Temples.
+type SandstoneChiseled struct {
```
Comment (Sandertv) [inline dragonfly/block/sandstone_chiseled.go:6]: ChiseledSandstone would make more sense.

File: dragonfly/item/flint.go [hunk 8]
```diff
@@ -0,0 +1,10 @@
+package item
+
+// Flint is a mineral obtained from gravel.
+type Flint struct {
+}
```
Comment (Sandertv) [inline dragonfly/item/flint.go:5]: You can put this on the same line as the opening `{`.

File: dragonfly/block/sandstone_cut.go [hunk 9]
```diff
@@ -0,0 +1,36 @@
+package block
+
+import "github.com/df-mc/dragonfly/dragonfly/item"
+
+// CutSandstone is a block which can be found in generated structures.
+type CutSandstone struct {
+	// Red specifies if the sandstone is red or not. ChiseledSandstone only has it's basic colour and red.
```
Comment (Sandertv) [inline dragonfly/block/sandstone_cut.go:7]: Typo: it's should be its.

File: dragonfly/block/gravel.go [hunk 10]
```diff
@@ -0,0 +1,27 @@
...
+type Gravel struct{}
+
+// BreakInfo ...
+func (g Gravel) BreakInfo() BreakInfo {
+	return BreakInfo{
+		Hardness:    0.6,
+		Harvestable: alwaysHarvestable,
+		Effective:   shovelEffective,
+		Drops:       simpleDrops(item.NewStack(g, 1)),
+		// TODO: Add Flint and drop it here.
```
Comment (Sandertv) [inline dragonfly/block/gravel.go:15]: You added flints in this pull request so make sure to drop them with the right chance.

**Comment by Sandertv** [comment]
Could you please make sure the names of the file match the type names? For example, `cut_sandstone` rather than `sandstone_cut`

**Comment by T14Raptor** [comment]
Closing this PR as stale. Most of these features were already implemented and there's conflicts.

**Pull request closed without merging**